scipy/optimize/_minimize.py

```diff
@@ -41,6 +41,10 @@
                     'l-bfgs-b', 'tnc', 'cobyla', 'slsqp', 'trust-constr',
                     'dogleg', 'trust-ncg', 'trust-exact', 'trust-krylov']
 
+# These methods may possibly pass jacobians or hessians to the callback
+MINIMIZE_METHODS_DERIV = ['bfgs', 'newton-cg', 'trust-constr',
+                        'dogleg', 'trust-ncg', 'trust-exact', 'trust-krylov']
```

Comment:
According to the `minimize` documentation, the list of methods that support `jac` is different from those that support `hess`. If this list is for `jac`, it also needs `'cg', 'l-bfgs-b', 'tnc', 'slsqp'`. Adding this should reveal that `cg` and `l-bfgs-b` should be including `jac` in `intermediate_result` but currently aren't. (`tnc` and `slsqp` are exempt because they do not support the `intermediate_result` version of the callback interface.)

The list says that `hess` is "Only for Newton-CG, dogleg, trust-ncg, trust-krylov, trust-exact and trust-constr."

```suggestion
MINIMIZE_METHODS_JAC = ['cg', 'bfgs', 'newton-cg', 'l-bfgs-b', 'tnc', 'slsqp',
                        'dogleg', 'trust-ncg',  'trust-krylov', 'trust-exact', 
                        'trust-constr']
MINIMIZE_METHODS_HESS = ['newton-cg', 'dogleg', 'trust-ncg',  'trust-krylov',
                         'trust-exact', 'trust-constr']
```